cocotb/triggers.py

```diff
@@ -815,7 +822,7 @@ async def with_timeout(trigger, timeout_time, timeout_unit=None):
         await with_timeout(First(coro, event.wait()), 100, 'ns')
 
     Args:
-        trigger (:class:`~cocotb.triggers.Trigger` or :class:`~cocotb.triggers.Waitable` or :class:`~cocotb.decorators.RunningTask`):
+        trigger (:class:`~cocotb.triggers.Trigger` or :class:`~cocotb.triggers.Waitable` or :class:`~cocotb.decorators.RunningTask` or :class:`types.CoroutineType`):
```

Comment:
```suggestion
        trigger (:class:`~cocotb.triggers.Trigger`, :class:`~cocotb.triggers.Waitable`, :class:`~cocotb.decorators.RunningTask`, or :class:`types.CoroutineType`):
```

Reply:
Is there a way to reference `coroutine` in https://docs.python.org/3/glossary.html#term-coroutine? I don't know if `:term:` would link it or not.

Reply:
A hyperlink. `:term:` seems to require a glossary be present, which is a good idea, but out of scope for this PR.

Reply:
``:term:`coroutine` `` or ``:term:`python:coroutine` `` should work even without a cocotb glossary (which we may want to have at some point, so the 2nd, more explicit, syntax is better), the magic is done with intersphinx. If you try this locally, you may have to run ``make clean`` in ``documentation/`` first.